test/test_pooling.py

```diff
@@ -520,6 +520,14 @@ def test_max_pool_size_with_connection_failure(self):
             self.assertNotIn('waiting for socket from pool',
                              str(context.exception))
 
+    def test_min_pool_size_always_positive(self):
```

Comment:
Rather than adding a new test I think we can re-use `test_return_socket_after_reset`: https://github.com/mongodb/mongo-python-driver/blob/b342990934b55370ebde40d20ba6423e035b1961/test/test_pooling.py#L291-L296

We just want to add some more assertions for active_sockets and probably operation_count too.

Reply:
Done. I'm not sure why you mean by adding assertions for operation_count as that appears to stay at 0 during the entirety of the test.